resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesTestConf.scala

```diff
@@ -35,7 +35,7 @@ object KubernetesTestConf {
   val RESOURCE_PREFIX = "prefix"
   val EXECUTOR_ID = "1"
 
-  private val DEFAULT_CONF = new SparkConf(false)
+  private val DEFAULT_CONF = new SparkConf(false).set(WAIT_FOR_APP_COMPLETION, false)
```

Comment:
Is this for reducing the amount of logging?

If so, isn't this a bit dangerous? Because this option doesn't really control logging, but whether to wait for the app. If there's too much log noise you could set the report interval to a really large value.

Maybe that doesn't matter for unit tests, but it feels a bit odd.

Reply:
https://github.com/apache/spark/blob/7912ab85a6fc086b814d93e7c71af1f50515517a/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala#L205

As the `spark.kubernetes.report.interval` is always positive, I guess if WAIT_FOR_APP_COMPLETION is true, the logging is inevitably on whether in a long or short interval.

I don't think this is dangerous, because in fact this only controls the logic of LoggingPodStatusWatcher before or after this change. 
For unit tests, this will only take effect in `k8s.submit.ClientSuite`, if it feels a bit odd to you, I can set this test by test in that test class

also cc @erikerlandson